src/ckeditor.jsx

```diff
@@ -139,7 +139,8 @@ CKEditor.propTypes = {
 	name: PropTypes.string,
 	style: PropTypes.object,
 	readOnly: PropTypes.bool,
-	onBeforeLoad: PropTypes.func
+	onBeforeLoad: PropTypes.func,
```

Comment:
I'm wondering if we should still keep `onBeforeLoad`. It seems to be not documented anywhere, on the other side we already see that someone could be using it: https://github.com/ckeditor/ckeditor4-react/issues/129#issuecomment-732939153

It would be good to keep API consistent with what we have in Vue and Angular, also this callback may create unexpected results described in https://github.com/ckeditor/ckeditor4-react/issues/129#issuecomment-732997609

Not sure if we should go with a deprecation message or just get rid of it and target this PR into a major release. @Comandeer do you have any thoughts on that issue?

Reply:
In [Color nrmalizer](https://github.com/ckeditor/ckeditor4/pull/4366) we add `@deprecated` even if PR targeting major release.

Reply:
>I'm wondering if we should still keep onBeforeLoad. It seems to be not documented anywhere, on the other side we already see that someone could be using it: #129 (comment)

>It would be good to keep API consistent with what we have in Vue and Angular, also this callback may create unexpected results described in #129 (comment)

Founded: https://github.com/ckeditor/ckeditor4-react/issues/47. Seems that the **initial driver** for adding `onBeforeLoad` callback is in contradiction with https://github.com/ckeditor/ckeditor4-react/issues/129#issuecomment-732997609

So we have `onBeforeLoad` - per instance call nad `onNamespaceLoaded` - call before all instances loaded and it's called once.

To me: those are two different behaviours which, however, may lead to confusion if used improperly. So maybe good documentation may help here? So I would stay with both callbacks and eventually mark `onBeforeLoad` with `@deprecated` if we would like to keep API consistent with others integrations.

Reply:
One note about deprecating stuff and versions - in CKEditor 4 we don't really follow semver - we have only `minor` and `major` releases there. CKEditor 4 `minor` is really a `patch` in semver nomenclature (bug fixes only) while CKEditor 4 `major` is `minor` in semver (new features). As for `semver` `major` which means breaking changes we really avoid such in CKEditor 4.

So according to semver (which we follow here), getting rid of `onBeforeLoad` entirely means breaking changes and it should be introduced in `major`. While [deprecating stuff should be done in `minor`](https://semver.org/#spec-item-7).